### PR TITLE
Add mentions and emojis suggestions to Home prompt box; fix RHP suggestions

### PR DESCRIPTION
**Pull request by GCyganek**
### Explanation of Change

Adds `@mention` and `:emoji` suggestions to the Home page Concierge prompt box. Extracted the logic shared by the report composer and message edit into a new `useComposerSuggestions` hook, reused by all three. Also fixes suggestions rendering behind the RHP by giving the suggestions portal an explicit z-index.

### Fixed Issues

$ https://github.com/Expensify/App/issues/99550
PROPOSAL: N/A

### Tests

Web: 

1. Open the Home page and focus the prompt box
2. Type `@` followed by a few letters of a known contact (e.g. `@exp`).
3. Verify that a mention suggestion list appears anchored to the prompt box. 
4. Press `Enter`.
5. Verify that the highlighted mention is inserted into the prompt box and that the prompt is **not** submitted.
6. Press `Enter` again (with no suggestion list open).
7. Verify that the prompt is submitted normally.
8. Go back to Home page and type `:smi` in the prompt box.
9. Verify that an emoji suggestion list appears, then press `Enter` and verify the emoji is inserted instead of the prompt being sent.
10. Type `@a` again to open the suggestions, then press `Escape`.
11. Verify that the suggestion list is dismissed.
12. Type `@a` again to open the suggestions, then click outside the prompt box so it loses focus.
13. Verify that the suggestion list is hidden.
14. Type a message with `@` to trigger mention suggestion list in a way that changes prompt box height with suggestion list open.
15. Verify that the height change of the prompt box does not hide the suggestion list and it's still opened 

RHP (web):
1. Open the Home page and click on the Help button in the top right corner to open Concierge in RHP
2. Type `@`.
3. Verify that a mention suggestion list appears.
4. Type `:smi` and verify that an emoji suggestion list appears
5. Verify that both suggestion list work correctly

Mobile app:
1. Open the Home page and tap on the prompt box
2. Type `@`.
3. Verify that a mention suggestion list appears anchored to the prompt box. 
4. Tap a contact from the list.
5. Verify that the selected mention is inserted into the prompt box.
6. Type `:smi` in the prompt box.
7. Verify that an emoji suggestion list appears, select an emoji and verify it's inserted into the prompt box
8. Type a message with `@` to trigger mention suggestion list in a way that changes prompt box height with suggestion list open.
9. Verify that the height change of the prompt box does not hide the suggestion list and it's still opened 
10. Verify that scrolling the screen with the suggestion list open closes suggestion list

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

Same as tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If new assets were added or existing ones were modified, I verified that:
    - [x] The assets are optimized and compressed (for SVG files, run `npm run compress-svg`)
    - [x] The assets load correctly across all supported platforms.
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] I added [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) for any new feature or bug fix in this PR to help automatically prevent regressions in this user flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/b874d2ad-5303-44e5-b09d-1fea6d7e104c

suggestions placement

https://github.com/user-attachments/assets/9cfff0f4-9f5f-4af4-a403-26ca357c2740

https://github.com/user-attachments/assets/01dd407b-0aaf-49f4-8d31-5668d344059c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/5b2a8f7c-7364-42b5-a124-ed45883dc3a1

suggestions placement:

https://github.com/user-attachments/assets/27b9f390-1157-4f69-9b9a-0aa5a7ca53b0

https://github.com/user-attachments/assets/bb38a932-f4e5-4f35-a478-150e1ee39db3

https://github.com/user-attachments/assets/9825a95d-083d-460d-99fa-1a5a68ed0e64

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/809ba689-bf1a-4701-a069-437867f62094

https://github.com/user-attachments/assets/064621ec-1460-48e7-8606-ad7400773297

https://github.com/user-attachments/assets/9a36961d-807e-4427-9092-7ce0bee5bb97

https://github.com/user-attachments/assets/a430716a-cb30-4974-aa72-31c327034af5

</details>

### CODE REVIEW

**Comment by GCyganek** [comment]
@codex please review

**Comment by dannymcclain** [comment]
This might be a silly question, but why would you ever need to @mention someone in that Concierge input?

**Comment by shawnborton** [comment]
My thinking is that it might be nice to do something like `Show me all violations from @danny last quarter`

**Comment by dannymcclain** [comment]
Ah yes! That makes perfect sense, thank you.

**Comment by shawnborton** [comment]
Hmm the send button is not active, I can't actually send this message:

CleanShot 2026-08-31 at 17 11 57@2x

File: src/pages/home/ForYouSection/ConciergePromptBox.tsx [hunk 1]
```diff
@@ -182,6 +206,11 @@ function ConciergePromptBox({isMenuVisible, setIsMenuVisible}: ConciergePromptBo
         if (canSkipTriggerHotkeys(shouldUseNarrowLayout, isKeyboardShown)) {
             return;
         }
+
+        if (suggestionsRef.current?.triggerHotkeyActions(event as unknown as KeyboardEvent)) {
+            return;
```
[the remark this answers is not in the document]
Comment (GCyganek) [reply src/pages/home/ForYouSection/ConciergePromptBox.tsx:206]: not true

**Comment by GCyganek** [comment]
@shawnborton are you still able to reproduce? I can't:

https://github.com/user-attachments/assets/c0638ff9-0f80-4203-8bd0-038b09ba44a9

**Comment by GCyganek** [comment]
Jest test failing on main, not caused by this PR, will merge main later when it's fixed

**Comment by shawnborton** [comment]
Hmm odd, I can't reproduce now - ignore me!

**Comment by ZhenjaHorbach** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified that the composer does not automatically focus or open the keyboard on mobile unless explicitly intended. This includes checking that returning the app from the background does not unexpectedly open the keyboard.
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: HybridApp
    - [ ] Android: mWeb Chrome
    - [ ] iOS: HybridApp
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] For any bug fix or new feature in this PR, I verified that sufficient [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) are included to prevent regressions in this flow.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: HybridApp</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: HybridApp</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

**Review by dylanexpensify — DISMISSED** [review]
good from a product perspective 👍

**Comment by ZhenjaHorbach** [comment]
@GCyganek 
Let's update the branch to the latest version of main to fix jest issues!

**Comment by ZhenjaHorbach** [comment]
I have a strange issue that in the mentions list, I have an extra @ at the end of the item

Снимок экрана — 2026-09-03 в 13 39 44

**Comment by ZhenjaHorbach** [comment]
But okay
This is because we additionally display the user's email
Although I'm not sure we even need to show this if there isn't enough space

Снимок экрана — 2026-09-03 в 13 41 46

**Comment by ZhenjaHorbach** [comment]
But we have a more critical issue on native apps
We show the suggestion box in the wrong position 

Снимок экрана — 2026-09-03 в 13 45 01

**Comment by ZhenjaHorbach** [comment]
Plus, we don't unmount the box when we leave the Home screen

https://github.com/user-attachments/assets/84af80b4-8fb7-4458-b306-0f0f708d832b

**Comment by GCyganek** [comment]
> But we have a more critical issue on native apps
> We show the suggestion box in the wrong position

Hmm I guess this is because the keyboard is not opened? I tested with keyboard opening on the simulator, but will check in a sec

**Comment by ZhenjaHorbach** [comment]
> > But we have a more critical issue on native apps
> > We show the suggestion box in the wrong position
> 
> Hmm I guess this is because the keyboard is not opened? I tested with keyboard opening on the simulator, but will check in a sec

Oh 
Indeed 😅
Good point!
With the keyboard, it's much better

**Comment by ZhenjaHorbach** [comment]
But now we have another issue with the Home concierge input
The suggestion box covers the input when we have 2 or more lines
We don't have this issue with the chat input 

https://github.com/user-attachments/assets/805897fa-fc2e-4513-a398-a748d315bcc7

**Comment by GCyganek** [comment]
Yeah I'm trying to see if I can improve this

**Comment by GCyganek** [comment]
I pushed a fix for the this:

> Plus, we don't unmount the box when we leave the Home screen


And I may have a solution for suggestions placement, but I need to test it with different platforms and edge cases to see if it's better

**Comment by GCyganek** [comment]
@ZhenjaHorbach I think I fixed it, I tested it on iOS and Android simulators, as well as on my Android device, but I may have missed some edge case. Changes shouldn't affect web. I will test some more and attach videos soon. I removed Android version of getBottomSuggestionPadding as from what I see with my fix it's not needed anymore

**Comment by GCyganek** [comment]
Added new videos to the PR description for Android and iOS to show how the suggestions placement works now

**Comment by GCyganek** [comment]
And to respond to this:

> But okay
> This is because we additionally display the user's email
> Although I'm not sure we even need to show this if there isn't enough space

I'd leave it as is as I see this is how it works on prod everywhere

**Comment by ZhenjaHorbach** [comment]
Rechecking!
@GCyganek 
And can you fix conflicts, please?

**Comment by GCyganek** [comment]
on it

**Comment by ZhenjaHorbach** [comment]
On mobile apps, we show the suggestion box under the input
Which is fine, I think because there is not enough space above the input
But I think we need to add scrolling if the input is outside the visible area
Since it breaks the suggestion box

https://github.com/user-attachments/assets/8969f485-9291-4eca-b597-6379174ef806

**Comment by ZhenjaHorbach** [comment]
And we have TS issue

**Comment by GCyganek** [comment]
> On mobile apps, we show the suggestion box under the input
> Which is fine, I think because there is not enough space above the input
> But I think we need to add scrolling if the input is outside the visible area
> Since it breaks the suggestion box

weird, it did not behave like this when I was testing it, will check again...

**Comment by GCyganek** [comment]
@ZhenjaHorbach what simulator and ios version did you test it on?

**Comment by ZhenjaHorbach** [comment]
> @ZhenjaHorbach what simulator and ios version did you test it on?

iPhone 17e and iOS 26.4

**Comment by GCyganek** [comment]
OMG this is on mWeb, not native app 🤦 checking how to fix it...

**Comment by GCyganek** [comment]
@ZhenjaHorbach should be fixed now, thanks:


https://github.com/user-attachments/assets/90497d46-a05b-4777-8b1e-73ad02c28038


https://github.com/user-attachments/assets/14f053cb-0d19-4f4d-b363-dfeb099607de

**Comment by ZhenjaHorbach** [comment]
Looks like on iOS web, the suggestion box is still broken 😅

https://github.com/user-attachments/assets/01d6ad75-2dbf-4d09-8256-d0701ef1de82

**Comment by ZhenjaHorbach** [comment]
And on iOS native, the emoji box overlaps content if displayed below the input

https://github.com/user-attachments/assets/77acab61-abd8-4a4a-a58b-3a9f336027e8

**Comment by ZhenjaHorbach** [comment]
But I think it's better to check on real devices
@dannymcclain @shawnborton @dylanexpensify 
Can you run adhoc builds, please?

**Comment by shawnborton** [comment]
Can do

**Comment by ZhenjaHorbach** [comment]
@GCyganek 
And I think the best fix is that after focusing on the input to scroll the screen to the fixed position so that the suggestion box always opens in one place

**Comment by ZhenjaHorbach** [comment]
On a real device also works not good 😅

https://github.com/user-attachments/assets/8b1933e9-d3d3-40c5-8faf-3d51c6b114fe

**Comment by GCyganek** [comment]
Ugh... let me check again, it was working fine for me yesterday on all devices 🙄

**Comment by GCyganek** [comment]
Ok I was able to reproduce, it works fine when `Good morning ${name}` line takes only one line, when I entered long first and last name it's broken for some time for some reason

**Comment by GCyganek** [comment]
@ZhenjaHorbach sorry for this, I tested mWeb many times on different browsers, should be working now:


https://github.com/user-attachments/assets/aadacc2c-e8af-428d-878f-c75e592d8551


https://github.com/user-attachments/assets/0c7a6cd6-c64c-4fa0-8621-d86d67a91b19

I'm looking into this one now:

> And on iOS native, the emoji box overlaps content if displayed below the input

**Comment by GCyganek** [comment]
> And on iOS native, the emoji box overlaps content if displayed below the input

fixed: 

https://github.com/user-attachments/assets/63933be0-b7e5-487c-b5a1-4b510e97a28b

File: src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx [hunk 2]
```diff
@@ -1,41 +1,77 @@
...
+    const hostRelativeBottom = hostFrameBottom === null ? 0 : bottom + keyboardHeight - (windowHeight - hostFrameBottom);
+    const isHostFrameMeasured = hostFrameBottom !== null;
+    const bottomPadding = getBottomSuggestionPadding(isMenuAbove);
+    const containerStyle = StyleUtils.getBaseAutoCompleteSuggestionContainerStyle({left, width, bottom: hostRelativeBottom + bottomPadding});
 
     if (!width) {
         return null;
     }
 
+    console.log({bottomPadding});
```
Comment (ZhenjaHorbach) [inline src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx]: I think we don't need this 😅

File: src/components/AutoCompleteSuggestions/getSuggestionsViewportBottom/index.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/**
+ * The menu's `bottom` counts up from the `html` box, so that box has to be measured with the same `getBoundingClientRect`
+ * the caret position comes from. A window height is not interchangeable: iOS scrolls the page to reveal the keyboard.
+ */
+function getSuggestionsViewportBottom(windowHeight: number): number {
+    return document.documentElement?.getBoundingClientRect().bottom ?? windowHeight;
```
Comment (ZhenjaHorbach) [inline src/components/AutoCompleteSuggestions/getSuggestionsViewportBottom/index.ts]: Since native apps don't have a DOM in any case, `getSuggestionsViewportBottom/index.native.ts` looks a bit redundant

File: src/components/AutoCompleteSuggestions/index.tsx [hunk 4]
```diff
@@ -185,14 +193,17 @@ function AutoCompleteSuggestions<TSuggestion>({measureParentContainerAndReportCu
         return null;
     }
 
+    console.log({isMenuAbove: containerState.isMenuAbove});
```
Comment (ZhenjaHorbach) [inline src/components/AutoCompleteSuggestions/index.tsx]: And this

File: src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx (hunk 2 above)
Comment (GCyganek) [reply src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx]: Yeah, sorry 😅 must've missed the eslint failure notification

File: src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx [hunk 5]
```diff
@@ -1,41 +1,75 @@
...
+    const hostFrameRef = useRef<View>(null);
+    const [hostFrameBottom, setHostFrameBottom] = useState<number | null>(null);
+
+    // Re-base `bottom` (measured from the window bottom, offset by the keyboard) onto the portal host's own frame, which can sit higher on screens with bottom-docked content.
+    const measureHostFrame = () => {
+        hostFrameRef.current?.measureInWindow((x, y, frameWidth, height) => setHostFrameBottom(y + height));
+    };
+    useEffect(measureHostFrame, [windowHeight, keyboardHeight, bottom]);
+
+    const hostRelativeBottom = hostFrameBottom === null ? 0 : bottom + keyboardHeight - (windowHeight - hostFrameBottom);
```
Comment (ZhenjaHorbach) [inline src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx]: Actually, keyboardHeight cancels out algebraically here (bottom already subtracts it) 
This rebasing is a no-op, doesn't actually offset for the keyboard

File: src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx [hunk 6]
```diff
@@ -1,41 +1,75 @@
...
-                />
-            </View>
+            {/* Zero-cost probe filling the portal host, so the host's position in the window is known before positioning the suggestions. */}
+            <View
+                ref={hostFrameRef}
+                pointerEvents="none"
+                style={styles.fullScreen}
+                onLayout={measureHostFrame}
+            />
+            {isHostFrameMeasured && (
```
Comment (ZhenjaHorbach) [inline src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx]: Minor comment
But the menu is now gated on isHostFrameMeasured: the first show can flicker/delay a frame
And before, we had old sync render

Comment (GCyganek) [reply src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx]: Yeah it's a deliberate choice to delay a frame here, it should not flicker though. I think it's better to delay a frame and be sure of the correct placement than stay with invalid placement

File: src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx (hunk 5 above)
Comment (GCyganek) [reply src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/index.native.tsx]: Yeah, it cancels out for the native app, I will do something about that

File: src/components/AutoCompleteSuggestions/getSuggestionsViewportBottom/index.ts (hunk 3 above)
Comment (GCyganek) [reply src/components/AutoCompleteSuggestions/getSuggestionsViewportBottom/index.ts]: I decided to keep native and web separate versions for better readability as I moved keyboardHeight subtraction to web version

**Comment by GCyganek** [comment]
@ZhenjaHorbach I simplified the logic a bit and tested all the platforms again, should be ok

**Comment by ZhenjaHorbach** [comment]
Looks like on Android 
The suggestion box is flickering

https://github.com/user-attachments/assets/5219a599-0515-4595-9eb9-870bd987bcd9

**Comment by ZhenjaHorbach** [comment]
But overall changes work well!

File: src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/getBottomSuggestionPadding/index.native.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+function getBottomSuggestionPadding(isMenuAbove: boolean): number {
```
Comment (ZhenjaHorbach) [inline src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/getBottomSuggestionPadding/index.native.ts:1]: Minor comment
And to me, using two files for this kind of logic is a small overcoding 😅

Can we use something like this?

```
function getBottomSuggestionPadding(isMenuAbove: boolean): number {
    return Platform.select({
        web: 6,
        default: isMenuAbove ? 30 : 0,
    });
}
```

Comment (GCyganek) [reply src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/getBottomSuggestionPadding/index.native.ts:1]: Isn't `Platform.select` breaking CONSISTENCY-1 AI reviewer rule?

Comment (ZhenjaHorbach) [reply src/components/AutoCompleteSuggestions/AutoCompleteSuggestionsPortal/getBottomSuggestionPadding/index.native.ts:1]: Oh
Okay then
I remember that platform specific files are preferable, but I thought it wasn't mandatory 😅
Especially in this case
Let's just leave everything as it is, then!

**Comment by GCyganek** [comment]
> Looks like on Android The suggestion box is flickering

Could you tell me what simulator & android version you are using on the video? I can't reproduce the flickering on my simulators and pixel 8 & android 16

**Comment by ZhenjaHorbach** [comment]
> > Looks like on Android The suggestion box is flickering
> 
> Could you tell me what simulator & android version you are using on the video? I can't reproduce the flickering on my simulators and pixel 8 & android 16

I have a Pixel 6 and Android 17

**Comment by GCyganek** [comment]
Ok I talked to @ZhenjaHorbach in DMs and we verified that flickering also happens on main in the concierge chat, so this is not something that was introduced on this PR, we can test on AdHoc to see if it's not only a simulator/dev build problem

**Comment by GCyganek** [comment]
@shawnborton @dylanexpensify could you trigger adhoc build here? :pray: thanks!

**Comment by shawnborton** [comment]
Doing it now 🚀

**Comment by GCyganek** [comment]
@ZhenjaHorbach I tested on two physical Android devices and couldn't reproduce flickering, maybe this is only simulator/dev build issue? This is not caused by this PR anyway so I think we can keep an eye on this and proceed

**Review by ZhenjaHorbach — APPROVED** [review]

**Comment by ZhenjaHorbach** [comment]
LGTM then!

**Review by puneetlath — APPROVED** [review]
Seems good to me from a code-perspective. Just confirming that @shawnborton @dylanexpensify you guys are good on the product side?

**Comment by shawnborton** [comment]
Hmm the list looks cut off at the bottom:
CleanShot 2026-09-10 at 20 42 55@2x


Which I am realizing is something that happens in regular chats too. So maybe ignore that, we can follow up and make it so that just the border of the popover cuts the overflow and not the awkward strip at the top/bottom. Otherwise good from me 👍

**Comment by ZhenjaHorbach** [comment]
Yeah
That’s an acceptable height for the suggestion box, which we didn't change in this PR
But probably, yes
We can improve it in the future!

**Comment by vincdargento** [comment]
Deploy Blocker https://github.com/Expensify/App/issues/100915 was identified to be related to this PR.

**Comment by unzig21** [comment]
Deploy Blocker https://github.com/Expensify/App/issues/100932 was identified to be related to this PR.

**Comment by jponikarchuk** [comment]
Deploy Blocker https://github.com/Expensify/App/issues/100934 was identified to be related to this PR.
